### PR TITLE
Add Makefile for quick deployment

### PR DESCRIPTION
**Pull request by mlimarenko**
## Description

Makefile-based quick deployment system with an interactive setup wizard. The new setup process simplifies LightRAG deployment by providing preset configurations (development, production, custom) and automating environment configuration, Docker Compose file generation, and service management.

The interactive wizard guides users through:
- Storage backend selection (PostgreSQL, Neo4j, MongoDB, Qdrant, Milvus, Redis, Memgraph)
- LLM provider configuration
- Docker service setup with customizable image tags
- Environment variable validation and backup
- Automatic docker-compose file generation

## Related Issues

N/A - This is a new feature to improve deployment experience and reduce setup time.

### Core Files
- **Makefile**: New Makefile with targets for `setup`, `setup-quick`, `setup-production`, `setup-validate`, `setup-backup`, and `setup-help`
- **scripts/setup/setup.sh**: Main interactive setup script (1218 lines) with wizard functionality

### Documentation
- **docs/InteractiveSetup.md**: New documentation for Makefile-based setup workflow
- **docs/DockerDeployment.md**: Updated with references to interactive setup and CPU mode configuration examples
- **env.example**: Enhanced with CPU mode configuration comments and additional deployment options

### Setup Scripts Library
- **scripts/setup/lib/file_ops.sh**: File operations utilities (backup, validation, etc.)
- **scripts/setup/lib/presets.sh**: Preset configuration definitions
- **scripts/setup/lib/prompts.sh**: Interactive prompt utilities
- **scripts/setup/lib/storage_requirements.sh**: Storage backend requirements and validation
- **scripts/setup/lib/validation.sh**: Environment variable and configuration validation

### Presets
- **scripts/setup/presets/development.sh**: Development preset (local JSON/NetworkX)
- **scripts/setup/presets/local.sh**: Local deployment preset
- **scripts/setup/presets/production.sh**: Production preset with security prompts

### Docker Compose Templates
- **scripts/setup/templates/**: Templates for all supported storage backends and services:
  - PostgreSQL, Neo4j, MongoDB, Qdrant, Milvus, Redis, Memgraph
  - vLLM rerank service (CPU and GPU variants)

## Checklist

- [x] Changes tested locally
- [x] Code reviewed
- [x] Documentation updated (if necessary)
- [ ] Unit tests added (if applicable)

### Key Features
1. **Quick Start**: `make setup-quick` provides a minimal-prompt development setup
2. **Production Ready**: `make setup-production` includes security and SSL configuration prompts
3. **Flexible**: Custom mode allows manual selection of each component
4. **Safe**: Automatic `.env` backup before modifications
5. **Validated**: Built-in validation of environment variables and configurations

### Backward Compatibility
- All existing deployment methods remain unchanged
- New Makefile targets are optional additions
- Existing `docker-compose.yml` is not modified by the wizard
- Generated compose files use separate names (e.g., `docker-compose.development.yml`)

### Usage Examples
# Quick development setup
make setup-quick

# Full production setup with security prompts
make setup-production

# Custom configuration wizard
make setup

# Validate existing configuration
make setup-validate

# Backup current .env
make setup-backup

### CODE REVIEW

**Comment by danielaskdd** [comment]
@codex review

File: scripts/setup/templates/vllm-rerank.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+  vllm-rerank:
+    image: vllm/vllm-openai:${VLLM_RERANK_IMAGE_TAG:-latest}
+    command: >
+      ${VLLM_RERANK_MODEL:-BAAI/bge-reranker-v2-m3}
+      --port ${VLLM_RERANK_PORT:-8000}
+      --dtype ${VLLM_RERANK_DTYPE:-float32}
```
[the remark this answers is not in the document]
Comment (danielaskdd) [reply scripts/setup/templates/vllm-rerank.yml:6]: The documentation in docs/DockerDeployment.md needs to be updated, as the --model parameter is unnecessary for vLLM."

**Comment by danielaskdd** [comment]
### Code review

Found 2 issues:

1. **vLLM template passes model as a bare positional arg, missing `--model` flag** — vLLM's `serve` command requires `--model` as a named argument, not a positional one. Both CPU and GPU templates will produce containers that fail to start. The documentation in the same PR correctly uses `--model BAAI/bge-reranker-v2-m3` (see `DockerDeployment.md` line 101), but the templates omit the flag.

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/templates/vllm-rerank.yml#L3-L6

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/templates/vllm-rerank-gpu.yml#L3-L6

2. **`check_storage_compatibility` returns 1 for informational warnings, breaking the development setup flow** — The function returns failure (`return 1`) whenever any warning exists (lines 95–100 of `validation.sh`). Since the development defaults (`JsonKVStorage`, `NetworkXStorage`, `JsonDocStatusStorage`) all individually trigger warning conditions, `select_storage_backends` will never break out of its loop cleanly for a standard dev setup — it always falls through to the "Proceed anyway?" confirm prompt. Warnings should print but still return 0 unless there is a hard incompatibility.

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/lib/validation.sh#L91-L103

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by danielaskdd** [comment]
### Code review

Found 5 issues:

1. **`gpus: all` is not valid Docker Compose v3 syntax** — the GPU template will silently fail to grant GPU access when using the modern `docker compose` plugin. The correct form uses `deploy.resources.reservations.devices`.

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/templates/vllm-rerank-gpu.yml#L14-L16

2. **Makefile calls `@bash` directly, breaking setup on macOS** — macOS ships Bash 3.2, but `setup.sh` uses `declare -A` (associative arrays) which require Bash 4+. Every `make setup*` target will fail on a stock Mac. The Makefile should resolve a Bash 4+ binary first (e.g. via a `SETUP_BASH ?= $(shell ...)` variable with fallbacks to `/opt/homebrew/bin/bash`, `/usr/local/bin/bash`).

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/Makefile#L41-L58

3. **Qdrant template defaults to a GPU-specific image tag** — `v1.16-gpu-nvidia` is a GPU build that will fail to start on CPU-only hosts (the majority of users). The default should be `latest` or a plain version tag like `v1.16.0`.

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/templates/qdrant.yml#L1-L3

4. **`check_storage_compatibility` returns `1` for informational warnings, treating them as hard errors** — the default development backends (NetworkX, NanoVectorDB, JsonKV) all trigger warning conditions, so the wizard always shows "Proceed anyway?" for a standard dev setup. Warnings should print to stderr and return `0`; only genuinely incompatible combinations should return `1`.

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/lib/validation.sh#L94-L101

5. **Neo4j and Memgraph templates both bind host port 7687** — if a user somehow selects both graph backends, `docker compose up` will fail with a port conflict. No guard in `check_storage_compatibility` covers this combination.

https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/templates/neo4j.yml#L7-L9
https://github.com/HKUDS/LightRAG/blob/a3e24adbe434312d9df76deffdd9297326c14863/scripts/setup/templates/memgraph.yml#L3-L5

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by danielaskdd** [comment]
@mlimarenko PR #2763 refactors the monolithic make wizard into modular env-base, storage, and server targets. Given LightRAG's complex underlying service dependencies and the dual deployment modes (Docker and non-Docker), ensuring full test coverage is challenging. While I have conducted extensive testing, edge cases may still exist. If your schedule permits, could you please pull the latest changes and verify if the configuration wizard functions as expected?
